### PR TITLE
fix: remember proxies from section data items

### LINKED ISSUE #13433: iOS: ListView with many AttributedStrings crashes
### I have searched and made sure there are no existing issues for the issue I am filing

- [X] I have searched the existing issues

### Description

Using a ListView with many attributedStrings will crash the iOS app.

### Expected Behavior

-

### Actual behavior

Hard crash

### Reproducible sample

```js
const win = Ti.UI.createWindow({});

function addListView() {
	var listView = Ti.UI.createListView({
		// footerView: mapView,
		templates: {
			test: {
				childTemplates: [{
					type: 'Ti.UI.View',
					childTemplates: [{
						type: 'Ti.UI.Label',
						bindId: 'label',
						properties: {
							color: 'black',
							bindId: 'label'
						}
					}],
					properties: {
						width: Ti.UI.FILL,
						height: 100,
						cardUseCompatPadding: true,
						backgroundColor: 'white'
					}
				}]
			}
		},
		defaultItemTemplate: 'test'
	});
	var section = Ti.UI.createListSection();
	var items = [];

	for (let i = 0; i < 700; i++) {
		var txt = "test text test text test text test text test text test text test text test text test text test text";
		var attr = Titanium.UI.createAttributedString({
			text: txt,
			attributes: [{
				type: Titanium.UI.ATTRIBUTE_PARAGRAPH_STYLE,
				range: [0, txt.length],
				value: {
					lineSpacing: 0,
					lineHeightMultiple: 0.8,
					lineBreakMode: Titanium.UI.ATTRIBUTE_LINE_BREAK_BY_WORD_WRAPPIN
				}
			}]
		});

		items.push({
			label: {
				attributedString: attr
			},
			template: 'test'
		})
	}

	section.setItems(items);
	listView.sections = [section];
	win.add(listView);
}

win.addEventListener('open', function() {
	addListView();
})
win.open();
```

### Steps to reproduce

* start app
* scroll down

### Platform

iOS

### SDK version you are using

10.1.1

### Alloy version you are using

_No response_

### PR DESCRIPTION
**Pull request by janvennemann**
A potential fix for https://github.com/tidev/titanium_mobile/issues/13433 we are currently testing. This remember all proxies that are inside a list data item to ensure they are properly protected from garbage collection between setting the `items` on a list section and the list view rerendering.

Fixes https://github.com/tidev/titanium_mobile/issues/13433

### CODE REVIEW

**Comment by m1ga** [comment]
nice, will check it later :+1: Had a quick test run with the old code but couldn't reproduce the crash with the latest xcode. I'll try to use the old app where I've found the crash for the first time, there I could reproduce it better

File: iphone/Classes/TiUIListSectionProxy.m [hunk 1]
```diff
@@ -137,11 +139,52 @@ - (void)setItems:(id)args withObject:(id)properties
...
+    for (NSString *key in item) {
+      if ([key isEqualToString:@"template"] || [key isEqualToString:@"properties"]) {
+        continue;
+      }
+      id itemPropertyValue = item[key];
+      if ([itemPropertyValue isKindOfClass:NSDictionary.class]) {
+        for (NSString *bindPropertyName in itemPropertyValue) {
+          id propertyValue = itemPropertyValue[bindPropertyName];
+          if ([propertyValue isKindOfClass:TiProxy.class]) {
+            [self forgetProxy:propertyValue];
```
Comment (hansemannn) [inline iphone/Classes/TiUIListSectionProxy.m:180]: Once validated: You could reuse the loop for both remember/forget to prevent duplicate code

**Comment by m1ga** [comment]
I can reproduce it with the old app (listview is a bit more complex images + attributedString) and once I have some items loaded and scroll up/down it will suddenly crash again. 
Using this PR I can scroll around for some minutes now without any crash :+1: I'll give it some more runs tomorrow but it's looking good. 
Any performances downsides using this approach? I see some nested for loops in there :smile: Not sure how often they'll run or if it depends on the amount of items

**Comment by hansemannn** [comment]
Regarding performance: As it's only done when setting items, it will not impact scroll performance. It *may* impact the speed of setting items, but I'll do some benchmarking there.

**Comment by janvennemann** [comment]
@m1ga As Hans already mentioned the for loops only run when setting new items. I did some time measurements, and the time spend in the for loops can be neglected compared to the total time it takes to set the items. Setting 5000 items on a section takes ~800ms on my iPhone X and only 4ms of that are spend in the for loops. Of course this might increase a little the more attributed string / proxies you use, but even then it shouldn't be noticeable.

BTW, here is the code that we were using to reproduce the crashes. It's a slightly modified version of your initial example, resetting the items every 500ms.

```js
const win = Ti.UI.createWindow({});

function addListView() {
	var listView = Ti.UI.createListView({
		templates: {
			test: {
				childTemplates: [{
					type: 'Ti.UI.View',
					childTemplates: [{
						type: 'Ti.UI.Label',
						bindId: 'label',
						properties: {
							color: 'black',
							bindId: 'label'
						}
					}],
					properties: {
						width: Ti.UI.FILL,
						height: 100,
						cardUseCompatPadding: true,
						backgroundColor: 'white'
					}
				}]
			}
		},
		defaultItemTemplate: 'test'
	});

	setInterval(() => {
		console.warn('REDRAW');

		var section = Ti.UI.createListSection();
		section.items = getItems();
		listView.sections = [section];
	}, 500);

	win.add(listView);
}

win.addEventListener('open', function () {
	addListView();
});
win.open();

function getItems() {
	const items = [];

	for (let i = 0; i < 700; i++) {
		var txt = 'test text test text test text test text test text test text test text test text test text test text' + new Date();
		var attr = Titanium.UI.createAttributedString({
			text: txt,
			attributes: [{
				type: Titanium.UI.ATTRIBUTE_PARAGRAPH_STYLE,
				range: [0, txt.length],
				value: {
					lineSpacing: 0,
					lineHeightMultiple: 0.8,
					lineBreakMode: Titanium.UI.ATTRIBUTE_LINE_BREAK_BY_WORD_WRAPPIN
				}
			}, {
				type: Titanium.UI.ATTRIBUTE_FOREGROUND_COLOR,
				range: [0, txt.length],
				value: 'red'
			}]
		});

		items.push({
			label: {
				attributedString: attr
			},
			template: 'test'
		});
	}

	return items;
}
```

**Review by hansemannn — APPROVED** [review]
Working fine for my test cases

**Review by m1ga — APPROVED** [review]
LGTM

tested the latest version with my app and the new test code. Both crashed without the patch and work fine with the patch :+1:

**Pull request merged**